clusterDeployer.py

```diff
@@ -119,8 +119,12 @@ def teardown(self) -> None:
         removed_macs = []
         names = [x.name for x in self._cc.all_vms()]
         ips = [x.ip for x in self._cc.all_vms()]
-        for e in q[-1][0][1:]:
-            if e.attrib["name"] in names or e.attrib["ip"] in ips:
+        dhcp = None
+        ip_tree = q.find('ip')
+        if ip_tree:
+            dhcp = ip_tree.find('dhcp')
+        for e in dhcp or []:
+            if e.get('name') in names or e.get('ip') in ips:
```

Comment:
```
ip_tree = q.find('ip')
dhcp = ip_tree.find('dhcp') if ip_tree else []
for e in dhcp:
```

Reply:
This could fail as dhcp =ip_tree.find('dhcp') could return Null (while ip_tree = q.find('ip') returned a proper element)
=> for e in dhcp will fail as not iterable
Could do:
ip_tree = q.find('ip')
dhcp = ip_tree.find('dhcp') if ip_tree 
for e in dhcp or []:

Reply:
```
ip_tree = q.find('ip')
dhcp = ip_tree.find('dhcp') if ip_tree
for e in dhcp or []:
```

looks good to me. Thanks for your last comment.